packages/server/modules/core/services/branch/management.ts

```diff
@@ -113,12 +135,31 @@ export const updateBranchAndNotifyFactory =
     }
 
     if (newBranch) {
-      await deps.addBranchUpdatedActivity({
-        update: input,
-        userId,
-        oldBranch: existingBranch,
-        newBranch
-      })
+      await Promise.all([
+        deps.eventEmit({
+          eventName: ModelEvents.Updated,
+          payload: {
+            update: input,
+            userId,
+            oldModel: existingBranch,
+            newModel: newBranch
+          }
+        }),
+        // TODO: Move to event bus listeners
```

Comment:
r u planning to move these to an event listener in a separate PR?

Reply:
ideally, we could remove the publishSub func from the dependencies

Reply:
@gjedlicska i'd rather have this be a separate task (or something to pick up within an unrelated task), cause its not really related to refactoring activityStream invocations. there's still a bunch I need to get through, and this is already better than what it was - activityStream invocations also triggering subscriptions

Reply:
if i wanna refactor this i need to now also introduce entirely new listeners only for subscriptions

Reply:
at that point, you could also clean up other events that happen during CRUD actions like "markBranchStreamAsUpdated", which strays even further from the original idea behind this task